### ISSUE TITLE
LLaDA-8B Model Ignores Attention Mask in Forward Process

### ISSUE THREAD

**Issue by Frankstein73** [ML-GSAI/LLaDA#89]
## 问题描述
LLaDA-8B 在 `forward` 时未正确处理 `attention_mask` 参数。对于含有 `pad_token` 的输入，无论是否在 `forward` 中传入 `attention_mask`，都会计算出相同的 `logits`，这与 LLaDA 的双向注意力机制不符。

## 复现代码
``` python
from transformers import AutoModel, AutoTokenizer
import torch

device = "cuda"
model = AutoModel.from_pretrained('/inspire/hdd/global_user/hezhengfu-240208120186/models/LLaDA-8B-Base', trust_remote_code=True, torch_dtype=torch.bfloat16).to(device).eval()
tokenizer = AutoTokenizer.from_pretrained('/inspire/hdd/global_user/hezhengfu-240208120186/models/LLaDA-8B-Base', trust_remote_code=True)

prompts = ["A", "A B"]

inputs = tokenizer(prompts, return_tensors="pt", padding=True, truncation=True)
input_ids = inputs.input_ids.to(device) # tensor([[32, 126081], [32, 413]])
attention_mask = inputs.attention_mask.to(device) # tensor([[1, 0], [1, 1]])

outputs_without_attention_mask = model(input_ids=input_ids)
outputs_with_attention_mask = model(input_ids=input_ids, attention_mask=attention_mask)

assert not torch.allclose(outputs_without_attention_mask.logits, outputs_with_attention_mask.logits)
# Due to the bio-attention of LLaDA, the logits of the first prompt should be different between the two outputs.
```
## 问题定位
在 [modeling_llada.py](https://huggingface.co/GSAI-ML/LLaDA-8B-Base/blob/main/modeling_llada.py#:~:text=v%2C-,attn_mask%3DNone%2C,-dropout_p%3D0.0) 计算 `attention` 的代码中，没有使用传入的 `attention_bias`。

**Comment by colinzhaoxp** [comment]
我也发现了这个问题，你有试过将这个修改为正确的attention_mask后来测试模型吗？

**Comment by NieShenRuc** [comment]
Thanks for your interest!

Since we didn't use attention masks during both pre-training and SFT processes, we simply set it to None for convenience. However, we have to admit that attention masks might be useful in certain scenarios, and I'm considering updating our code.

**Comment by Kamichanw** [comment]
Hi, I've fixed this bug in this [PR](https://huggingface.co/GSAI-ML/LLaDA-8B-Instruct/discussions/13), could you test it? @Frankstein73
